src/doc/rustc/src/codegen-options/index.md

```diff
@@ -209,6 +209,19 @@ Note that while the `-C instrument-coverage` option is stable, the profile data
 format produced by the resulting instrumentation may change, and may not work
 with coverage tools other than those built and shipped with the compiler.
 
+## jump-tables
+
+This option is used to allow or prevent the codegen backend from creating
+jump tables when lowering switches.
+
+* `y`, `yes`, `on`, `true` or no value: allow jump tables (the default).
+* `n`, `no`, `off` or `false`: disable jump tables.
+
+Disabling jump tables can be used to help provide protection against
+jump-oriented-programming (JOP) attacks, such as with the linux kernel's [IBT].
```

Comment:
Discussion [GUARANTEE 3/3]: If the flag is intended to be a hint, then this sentence can be a bit misleading, because we may not always guarantee it. We may want to slightly caveat this wording to not convey a "false promise" so to speak.

Or, if a user do want such protection, then do they need to enforce it over the whole crate graph?

Reply:
The flag should be guarantee anything compiled with jump tables turned off does not contain jump tables. It would be up to the user to enforce this dependency beyond the individual crate.

I think it might be better to remove this statement entirely. Maybe it would be better rewriten as:
```
Disabling jump tables can be used to help provide protection against
jump-oriented-programming (JOP) attacks. However, this option makes
no guarantee any precompiled or external dependencies are compiled 
with or without jump tables.
```